### ISSUE TITLE
bootloader: fatal error during startup after failed firmware update

### ISSUE THREAD

**Issue by mcudev** [trezor/trezor-core#109]
To simulate a failed firmware update, which leaves the device stalled and displaying only a percentage of the circular progress indicator, I started an update and hit Ctrl+c:
```
python-trezor$ python2 trezorctl firmware_update -f ../trezor-core/build/firmware/firmware.bin
Please confirm action on device...
<wait for installing firmware screen to display and start working>
^C
Aborted!
```

I then unplugged and re-plugged the device and got a fatal error:
```
msg: invalid firmware hash
file: embed/bootloader/main.c:319
rev: 37f6987
```

In this case, you probably want to start the bootloader's USB loop and treat it as a firmware not present scenario.

Also, from a user perspective, they should be told to simply try again when the firmware installation stalls (without unplugging). That is, this works without unplug/replug:

```
python-trezor$ python2 trezorctl firmware_update -f ../trezor-core/build/firmware/firmware.bin
Please confirm action on device...
<wait for installing firmware screen to display and start working>
^C
Aborted!
python-trezor$ python2 trezorctl firmware_update -f ../trezor-core/build/firmware/firmware.bin
Please confirm action on device...
True
```

**Comment by mcudev** [comment]
Closing, because after thinking about this, maybe nothing needs to be done here since it's recoverable. The user can always unplug/replug and do the screen touch thing to get into the bootloader USB loop.

**Comment by prusnak** [comment]
Right, my colleague encountered the issue as well. Fixed in 34c6fb987a16ecce93fdb786116787dc5be237aa

**Comment by mcudev** [comment]
ok, yeah, this is nicer / more user friendly
